hermes-api/src/main/java/pl/allegro/tech/hermes/api/OfflineRetransmissionRequest.java

```diff
@@ -8,27 +8,54 @@
 import pl.allegro.tech.hermes.api.jackson.InstantIsoSerializer;
 
 import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
 
 public class OfflineRetransmissionRequest {
+
+    private static final List<DateTimeFormatter> formatters = List.of(
+            DateTimeFormatter.ISO_INSTANT,
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm'Z'").withZone(ZoneId.of("UTC"))
+    );
+
     @NotEmpty
     private final String sourceTopic;
     @NotEmpty
     private final String targetTopic;
     @NotNull
-    private final Instant startTimestamp;
+    private Instant startTimestamp;
     @NotNull
-    private final Instant endTimestamp;
+    private Instant endTimestamp;
 
     @JsonCreator
     public OfflineRetransmissionRequest(
             @JsonProperty("sourceTopic") String sourceTopic,
             @JsonProperty("targetTopic") String targetTopic,
-            @JsonProperty("startTimestamp") Instant startTimestamp,
-            @JsonProperty("endTimestamp") Instant endTimestamp) {
+            @JsonProperty("startTimestamp") String startTimestamp,
+            @JsonProperty("endTimestamp") String endTimestamp) {
         this.sourceTopic = sourceTopic;
         this.targetTopic = targetTopic;
-        this.startTimestamp = startTimestamp;
-        this.endTimestamp = endTimestamp;
+        initializeTimestamps(startTimestamp, endTimestamp);
+    }
+
+    private void initializeTimestamps(String startTimestamp, String endTimestamp) {
+        if (startTimestamp == null || endTimestamp == null) {
+            this.startTimestamp = null;
```

Comment:
Aren't lines 45 and 46 redundant?

Reply:
Thats right, we could just assign one of them and the validation will fail, but I decided to do it for both to be more consistent and clear.